out_writeapi_test.go

```diff
@@ -38,3 +47,244 @@ func TestFLBPluginRegister(t *testing.T) {
+	CloseFunc      func() error
+	FinalizeFunc   func(ctx context.Context, opts ...gax.CallOption) (int64, error)
+	FlushRowsFunc  func(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error)
+	StreamNameFunc func() string
+}
+
+func (m *MockManagedStream) AppendRows(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+	return m.AppendRowsFunc(ctx, data, opts...)
+}
+
+func (m *MockManagedStream) Finalize(ctx context.Context, opts ...gax.CallOption) (int64, error) {
+	return m.FinalizeFunc(ctx, opts...)
+}
+
+func (m *MockManagedStream) FlushRows(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error) {
+	return m.FlushRowsFunc(ctx, offset, opts...)
+}
+
+func (m *MockManagedStream) StreamName() string {
+	return m.StreamNameFunc()
+}
+
+func (m *MockManagedStream) Close() error {
+	return m.managedstream.Close()
+}
+
+func TestFLBPluginFlushCtx(t *testing.T) {
+	checks := new(StreamChecks)
+	var setID int
+	mockClient := &MockManagedWriterClient{
+		NewManagedStreamFunc: func(ctx context.Context, opts ...managedwriter.WriterOption) (*managedwriter.ManagedStream, error) {
+			return nil, nil
+
+		},
+		GetWriteStreamFunc: func(ctx context.Context, req *storagepb.GetWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+			return &storagepb.WriteStream{
+				Name: "mockstream",
+				TableSchema: &storagepb.TableSchema{
+					Fields: []*storagepb.TableFieldSchema{
+						{Name: "Time", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+						{Name: "Text", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+					},
+				},
+			}, nil
+		},
+	}
+
+	originalFunc := getClient
+	getClient = func(ctx context.Context, projectID string) (ManagedWriterClient, error) {
+		log.Println("Mock ManagedWriterClient called")
+		return mockClient, nil
+	}
+	defer func() { getClient = originalFunc }()
+
+	res := []bool{}
+	md, _ := getDescriptors(ms_ctx, mockClient, "dummy", "dummy", "dummy")
+	mockMS := &MockManagedStream{
+		AppendRowsFunc: func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+			checks.appendRows++
+
+			var combinedData []byte
+			for _, tempData := range data {
+				combinedData = append(combinedData, tempData...)
+			}
+
+			message := dynamicpb.NewMessage(md)
+			err := proto.Unmarshal(combinedData, message)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to unmarshal")
+			}
+
+			textField := message.Get(md.Fields().ByJSONName("Text"))
+			timeField := message.Get(md.Fields().ByJSONName("Time"))
+
+			if textField.String() != "FOO" && textField.String() != "BAR" {
+				return nil, fmt.Errorf("Wrong Text Value: %s", textField.String())
+			}
+			log.Printf("String: %s", textField.String())
+			if timeField.String() != "000" {
```

Comment:
Same here, can we just perform a verification check here?

Reply:
Done.

Reply:
I mean can we directly use an "assert" statement here to fail the test?